### PR TITLE
Add av1 encode sample code

### PR DESCRIPTION
**Pull request by leyu-yao**
Add av1e sample code

### CODE REVIEW

**Comment by yellowdoge** [comment]
Hi! Can we get this patch reviewed and landed?

(Also I'm mcasas@chromium.org :)

**Comment by rosetta-jpn** [comment]
Gentle ping for review?

**Comment by sushma08** [comment]
@XinfengZhang can you please help review this patch or add necessary folks who have code merge rights? Thank you.

**Comment by sivileri** [comment]
+ @dvrogozh FYI

File: encode/av1encode.c [hunk 1]
```diff
@@ -0,0 +1,2964 @@
...
+#include <sys/stat.h>
+#include <sys/time.h>
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <CHECK_CONDITION.h>
+#include <pthread.h>
+#include <errno.h>
+#include <math.h>
+#include <va/va.h>
+#include <va/va_enc_av1.h>
```
Comment (XinfengZhang) [inline encode/av1encode.c:19]: redundant?

File: encode/av1encode.c [hunk 2]
```diff
@@ -0,0 +1,2964 @@
...
+}
+
+static void
+fill_sps_header()
+{
+    memset(&sh, 0, sizeof(sh));
+
+    sh.seq_level_idx[0] = ips.level;
+
+    sh.frame_width_bits = 15;
```
Comment (XinfengZhang) [inline encode/av1encode.c:1103]: all these if fix value?

File: encode/av1encode.c [hunk 3]
```diff
@@ -0,0 +1,2964 @@
...
+    fh.loop_filter_params.loop_filter_delta_update = 0;
+
+    fh.TxMode = TX_MODE_SELECT;
+    fh.reference_select = ips.LDB ? 1 : 0;
+    fh.skipModeAllowed = 0;
+    fh.skipModeFrame[0] = 0;
+    fh.skipModeFrame[1] = 0;
+    fh.allow_warped_motion = 0;
+    fh.reduced_tx_set = 1;
+
```
Comment (XinfengZhang) [inline encode/av1encode.c:1228]: why too many blank lines

File: encode/av1encode.c [hunk 4]
```diff
@@ -0,0 +1,2964 @@
...
+    bitstream_free(&bs);
+}
+
+static void
+render_TD()
+{
+    // OBU_TEMPORAL_DELIMITER
+    bitstream bs;
+    bitstream_start(&bs);
+    put_ui(&bs, 0x12, 8);
```
Comment (XinfengZhang) [inline encode/av1encode.c:1336]: fix length fix value？

Comment (leyu-yao) [reply encode/av1encode.c:1336]: This is for OBU_TEMPORAL_DELIMITER. This obu does not have payload, so length is not needed. Value is fixed and calculated directly from av1 spec.

File: encode/av1encode.c (hunk 2 above)
Comment (leyu-yao) [reply encode/av1encode.c:1103]: Suggest using these default values.

File: encode/av1encode.c (hunk 3 above)
Comment (leyu-yao) [reply encode/av1encode.c:1228]: Thanks, will updated

File: encode/av1encode.c (hunk 1 above)
Comment (leyu-yao) [reply encode/av1encode.c:19]: Av1 has some specific structures defined in va_enc_av1.h

**Review by XinfengZhang — APPROVED** [review]

**Comment by yellowdoge** [comment]
\o/

**Pull request merged**